app/batid/services/candidate.py

```diff
@@ -215,15 +222,16 @@ def sort_handle_null(lst):
 
         bdg_addresses = sort_handle_null(bdg.addresses_id)
         candidate_addresses = sort_handle_null(self.candidate.address_keys)
-        if bdg_addresses != candidate_addresses:
-            has_changed = True
+        if bdg_addresses != candidate_addresses and candidate_addresses:
```

Comment:
Je pense que cette comparaison n'est pas suffisante. Imaginons la situation où:

- le bâtiment contient les adresses ["one", "two"]
- le candidat contient les adresses ["one"]

Dans cette situation les deux scénarios de traitement seraient:

- 1: on retire l'adresse `two` du bâtiment car le candidat nous dit qu'il n'y qu'une seule adresse (`one`) pour ce bâtiment
- 2: on ne change pas le bâtiment car on considère que les candidats ne peuvent qu'apporter de nouvelles adresses (même ce cas mérite d'être affiné mais c'est un autre sujet) et non en supprimer.

Selon moi, on doit opter pour le scénario 2. Il est quasi certain qu'à terme les adresses dans le RNB seront de meilleures qualité que les adresses issues d'une autre base. Supprimer des adresses du RNB reviendrait à écarter le travail des contributeurs.

Avec le code de la PR, on aurait *quand même* une mise à jour car `bdg_addresses != candidate_addresses and candidate_addresses`.

Il faut plutôt vérifier si le candidat apporte une ou des adresses qui ne sont pas déjà dans le bâtiment (et même dans l'idéal, qui ne sont pas dans une des versions de l'historique du bâtiment).

Reply:
very true, merci